### PR TITLE
nixos/generic-extlinux-compatible: implement append secrets to initrd…

### PR DESCRIPTION
**Pull request by digtail**
… functionality

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x;] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/doc/manual/release-notes/rl-2311.section.md [hunk 1]
```diff
@@ -76,6 +76,8 @@
 
 - `services.nginx` gained a `defaultListen` option at server-level with support for PROXY protocol listeners, also `proxyProtocol` is now exposed in `services.nginx.virtualHosts.<name>.listen` option. It is now possible to run PROXY listeners and non-PROXY listeners at a server-level, see [#213510](https://github.com/NixOS/nixpkgs/pull/213510/) for more details.
 
+- `genric-extlinux-compatible` bootloader (and raspberry pi with uboot) supports appending secrets to the initramfs
```
Comment (yu-re-ka) [inline nixos/doc/manual/release-notes/rl-2311.section.md:79]:
```suggestion
- `generic-extlinux-compatible` bootloader (and raspberry pi with uboot) supports appending secrets to the initramfs
```

**Review by yu-re-ka — COMMENTED** [review]
Generally a good idea if this works with u-boot it is valuable for people using ssh in initrd etc.
Although since it is _generic_ extlinux compatible, can we really make assumptions about the bootloader? Do some bootloaders _not_ support multiple cpio archives being concatenated together as initrd? I have no idea.
My suggestion is to make supportsInitrdSecrets mkDefault false and then let the user set it in generic-extlinux-compatible.

**Comment by vcunat** [comment]
Channel is [blocked](https://hydra.nixos.org/build/236229201).  I _think_ this is the PR which caused it
```
$ nix build -f nixos/release.nix sd_image.aarch64-linux
error: builder for '/nix/store/mvxrg4vdp1w0j2zhq7f3fifz399swdck-ext4-fs.img.zst.drv' failed with exit code 1;
       last 1 log lines:
       > chown: changing ownership of './files/boot/nixos/qlyaznkh3aqd123l3h0s24pvz8fbbpcj-nixos-system-nixos-23.11pre130979.gfedcba-initramfs-with-secrets': Invalid argument
       For full logs, run 'nix log /nix/store/mvxrg4vdp1w0j2zhq7f3fifz399swdck-ext4-fs.img.zst.drv'.
error: 1 dependencies of derivation '/nix/store/yggdzh3z28pl7g2v1g6icbf5ima59f82-nixos-sd-image-23.11pre130979.gfedcba-aarch64-linux.img.drv' failed to build
```

**Comment by vcunat** [comment]
Confirmed by bisection that 94e939985b7730fd74b1c2e03292661734b490f0 is the culprit.

**Pull request merged**